pynumdiff/optimize/_optimize.py

```diff
@@ -101,42 +109,42 @@ def _objective_function(point, func, x, dt, singleton_params, search_space_types
     """
     point_params = {k:(v if search_space_types[k] == float else 
-                int(np.round(v)) if search_space_types[k] == int else
-                v > 0.5) for k,v in zip(search_space_types, point)} # point -> dict
+                int(np.round(v))) for k,v in zip(search_space_types, point)} # point -> dict
     # add back in the singletons we're not searching over
-    try: x_hat, dxdt_hat = func(x, dt, **point_params, **singleton_params) # estimate x and dxdt
+    try: x_hat, dxdt_hat = func(x, dt, **point_params, **singleton_params, **categorical_params) # estimate x and dxdt
     except np.linalg.LinAlgError: return 1000000000 # some methods can fail numerically
 
     # evaluate estimate
     if dxdt_truth is not None: # then minimize ||dxdt_hat - dxdt_truth||^2
         if metric == 'rmse':
-            rms_rec_x, rms_x, rms_dxdt = evaluate.metrics(x, dt, x_hat, dxdt_hat, dxdt_truth=dxdt_truth, padding=padding)
+            rms_rec_x, rms_x, rms_dxdt = evaluate.rmse(x, dt, x_hat, dxdt_hat, dxdt_truth=dxdt_truth, padding=padding)
             return rms_dxdt
         elif metric == 'error_correlation':
             return evaluate.error_correlation(dxdt_hat, dxdt_truth, padding=padding)
     else: # then minimize [ || integral(dxdt_hat) - x||^2 + gamma*TV(dxdt_hat) ]
-        rms_rec_x, rms_x, rms_dxdt = evaluate.metrics(x, dt, x_hat, dxdt_hat, dxdt_truth=None, padding=padding)
+        rms_rec_x, rms_x, rms_dxdt = evaluate.rmse(x, dt, x_hat, dxdt_hat, dxdt_truth=None, padding=padding)
         return rms_rec_x + tvgamma*evaluate.total_variation(dxdt_hat, padding=padding)
 
 
-def optimize(func, x, dt, search_space={}, dxdt_truth=None, tvgamma=1e-2, padding=0, metric='rmse',
-    opt_method='Nelder-Mead', maxiter=10):
+def optimize(func, x, dt, dxdt_truth=None, tvgamma=1e-2, search_space_updates={}, metric='rmse',
+    padding=0, opt_method='Nelder-Mead', maxiter=10):
     """Find the optimal parameters for a given differentiation method.
 
     :param function func: differentiation method to optimize parameters for, e.g. linear_model.savgoldiff
     :param np.array[float] x: data to differentiate
     :param float dt: step size
-    :param dict search_space: function parameter settings to use as initial starting points in optimization,
-                    structured as :code:`{param1:[values], param2:[values], param3:value, ...}`. The search space
-                    is the Cartesian product. If left None, a default search space of initial values is used.
     :param np.array[float] dxdt_truth: actual time series of the derivative of x, if known
     :param float tvgamma: Only used if :code:`dxdt_truth` is given. Regularization value used to select for parameters
                     that yield a smooth derivative. Larger value results in a smoother derivative.
+    :param dict search_space_updates: At the top of :code:`_optimize.py`, each method has a search space of parameters
```

Comment:
I renamed this param, because the old name was a little too grandiose for what it is.